lib/path/default.nix

```diff
@@ -92,8 +95,114 @@ let
     # An empty string is not a valid relative path, so we need to return a `.` when we have no components
     (if components == [] then "." else concatStringsSep "/" components);
 
+  # Takes a Nix path value and deconstructs it into:
+  # - root: The filesystem root of the path, generally `/`
+  # - components: All the path components of the path
```

Comment:
Well, [POSIX calls it path components](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_272), [RFC 3986 (URI) calls it segments](https://www.rfc-editor.org/rfc/rfc3986#section-3.3), though I don't think that's the same as unix paths. As for libraries:
- Python: [segments](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath)
- Nodejs: [segments](https://nodejs.org/api/path.html#pathresolvepaths)
- Rust: [components](https://doc.rust-lang.org/std/path/struct.Path.html#method.components)
- C++: weakly [components](https://en.cppreference.com/w/cpp/filesystem/path/remove_filename)
- Java: [name element](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#iterator--)
- Objective-C/Swift: [components](https://developer.apple.com/documentation/foundation/nsstring/1414489-pathcomponents)
- R: [components](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/normalizePath)

It looks like "segments" is more Web and URL related in general, whereas "components" is more focused on local paths. I'd say "components" should be fine here

Reply:
Alright, the word itself has some degree of freedom within reasonable proximity to the intended meaning. It could as well be part, item, node, piece, whatever. It should stay close to what's in common use though, therefore component and segment are valid candidates.

I don't think use of a term in context of Unix paths vs. URIs is a meaningful distinction, they describe the same concept from slightly different angles, and therefore in consider it a weak argument.

A stronger argument would be about what the word originally means, and if two words fit the meaning equally well, which one is easier to write and pronounce.

And then, indeed since [segment](https://en.m.wiktionary.org/wiki/segment) *also* refers to a "piece cut out or off" (which could be any subsequence of the items comprising a path) it's less suitable than [component](https://en.m.wiktionary.org/wiki/component), which refers to the things that are "put together" to make the whole.